app/src/main/java/doit/study/droid/QuestionFragment.java

```diff
@@ -44,16 +46,19 @@
     private LinearLayout mvAnswersLayout;
     private TextView mvRight;
     private TextView mvWrong;
+    private Button mvNextButton;
 
 
     // Factory method
-    public static QuestionFragment newInstance(int position) {
+    public static QuestionFragment newInstance(int position, int questionId, AnswerCheckListener listener) {
         if (DEBUG) Log.i("NSA", "newInstance "+position);
         // add Bundle args if needed here before returning new instance of this class
         QuestionFragment fragment = new QuestionFragment();
         Bundle bundle = new Bundle();
         bundle.putInt(ID_KEY, position);
         fragment.setArguments(bundle);
+        fragment.mQuestionId = questionId;
+        fragment.mAnswerCheckListener = listener;
```

Comment:
You can not do anything with low memory killer.
Another example incoming call puts the app in a background and system can optimise memory by killing... So in examples, you often can see static method with packing arguments in a bundle.
Do not save listener use callbacks.